daemon/mgr/container.go

```diff
@@ -1175,6 +1179,47 @@ func (mgr *ContainerManager) Restart(ctx context.Context, name string, timeout i
 	return mgr.start(ctx, c, "")
 }
 
+// DisconnectContainerFromNetwork disconnects the given container from
+// given network
+func (mgr *ContainerManager) DisconnectContainerFromNetwork(ctx context.Context, containerName, networkName string, force bool) error {
+	c, err := mgr.container(containerName)
+	if err != nil {
+		// TODO(ziren): if force is true, force delete endpoint
+		return err
+	}
+
+	// Get network
+	network, err := mgr.NetworkMgr.Get(ctx, networkName)
+	if err != nil {
+		return fmt.Errorf("failed to disconnect container %s from %s: get network failed: %v", c.Name(), networkName, err)
+	}
+
+	if c.meta.NetworkSettings != nil {
```

Comment:
I do not know if the logic is correct. While I insist the code could be simplified, like 
```
if c.meta.NetworkSettings == nil {
    return nil
}
epConfig, ok := c.meta.NetworkSettings.Networks[network.Name]
if !ok {
    return fmt.Errorf("failed to disconnect container from network: container %s not attach to %s", c.Name(), networkName)
}

...........

```
Using return fast to reduce ident of code.

Reply:
Agree !